File: src/mVasu.Api.Contracts/TaskDetailDto.cs
namespace mVasu.Api.Contracts;

/// <summary>
/// Full per-task detail returned by <c>GET /api/tasks/:id</c>. Maps to
/// SCREENS.md §04 (Tehtävän tarkka näkymä) and BEHAVIOR.md §5
/// (TaskDetailState shape). Phase 1 returns the same hand-curated mock
/// dataset as <see cref="ITaskQueryService.ListAsync"/>; Phase 2 will
/// resolve the source XAF entity by <see cref="EntityRef"/>.
/// </summary>
public sealed record TaskDetailDto(
    string Id,
    /// <summary>One of <see cref="TaskTypeNames"/>.</summary>
    string Type,
    /// <summary>
    /// Caption shown in the topbar — UserTaskType.Name for
    /// xVasuSecuritySystemUserTask rows, otherwise the localised
    /// label for the static <see cref="Type"/>.
    /// </summary>
    string TypeLabel,
    /// <summary>One of <see cref="TaskAccents"/>.</summary>
    string Accent,
    /// <summary>
    /// Hero pill text — pre-localised by the server (e.g.
    /// "Tänään klo 09:00 · 3 h kuluttua"). Frontend renders verbatim.
    /// </summary>
    string TimeContext,
    string Title,
    string? Subtitle,
    TaskCustomerDto? Customer,
    IReadOnlyList<TaskRowActionDto> Actions,
    TaskNoteDto? Note,
    TaskActionDto PrimaryCta,
    TaskEntityRefDto EntityRef);

/// <summary>Customer block on the detail screen — avatar + name + phone + email.</summary>
public sealed record TaskCustomerDto(
    string Id,
    string Name,
    string Initials,
    string? Phone,
    string? Email);

/// <summary>
/// Row-action shown in the TOIMINNOT list. Distinct from inline-card
/// <see cref="TaskActionDto"/>s because detail rows can require
/// confirmation and never carry a primary flag.
/// </summary>
public sealed record TaskRowActionDto(
    string Id,
    string Label,
    /// <summary>One of <see cref="TaskRowActionKinds"/>.</summary>
    string Kind,
    bool Destructive,
    /// <summary>
    /// Confirmation copy — when null the action runs immediately. The
    /// frontend's reason-picker takes over when <see cref="Kind"/> is
    /// <c>cancel</c> regardless of <see cref="Confirm"/>.
    /// </summary>
    TaskConfirmDto? Confirm,
    string? Href);

/// <summary>Inline confirm dialog copy attached to a row action.</summary>
public sealed record TaskConfirmDto(string Title, string Body);

/// <summary>Inline-editable note. <see cref="UpdatedAt"/> is server-assigned.</summary>
public sealed record TaskNoteDto(
    string Id,
    string Body,
    DateTimeOffset UpdatedAt);

/// <summary>Stable identifiers for the TOIMINNOT row dispatch.</summary>
public static class TaskRowActionKinds
{
    public const string Navigate    = "navigate";
    public const string External    = "external";
    public const string MarkDone    = "mark-done";
    public const string Cancel      = "cancel";
    public const string CreateOffer = "create-offer";
}

File: src/mVasu.Pwa/src/app/core/models/task-detail.dto.ts
import type { TaskAccent, TaskAction, TaskEntityRef, TaskType } from './task.dto';

/**
 * Detail-view shape for a single task. Mirrors BEHAVIOR.md §5 plus
 * NAVIGATION.md §3d (canonical xVasuSecuritySystemUserTask DetailView
 * layout). The aggregated `/api/tasks/:id` endpoint will produce this
 * shape at runtime; for now it is built from the local mock fixture.
 */
export interface TaskDetail {
  readonly id: string;
  readonly type: TaskType;
  readonly typeLabel: string;
  readonly accent: TaskAccent;

  /** "Tänään klo 09:00 · 3 h kuluttua" */
  readonly timeContext: string;
  /** Hero title — Huoneisto.Osoite ?? Kohde.Nimi ?? Subject */
  readonly title: string;
  /** Hero subtitle — "2 h, 47 m² · vapaa 1.7. alkaen" */
  readonly subtitle?: string;

  readonly customer?: TaskCustomer;
  readonly actions: ReadonlyArray<TaskRowAction>;
  readonly note: TaskNote | null;
  readonly primaryCta: TaskAction;

  readonly entityRef: TaskEntityRef;
}

export interface TaskCustomer {
  readonly id: string;
  readonly name: string;
  readonly initials: string;
  readonly phone?: string;
  readonly email?: string;
}

export type TaskRowActionKind =
  | 'navigate'
  | 'external'
  | 'mark-done'
  | 'cancel'
  | 'create-offer';

export interface TaskRowAction {
  readonly id: string;
  readonly label: string;
  readonly kind: TaskRowActionKind;
  readonly destructive?: boolean;
  /** Confirmation modal copy when set; missing = action runs immediately. */
  readonly confirm?: { readonly title: string; readonly body: string };
  readonly href?: string;
}

export interface TaskNote {
  readonly id: string;
  readonly body: string;
  readonly updatedAt: string;
}
